### DISCUSSION TITLE
useFieldArray together with watch

### DISCUSSION THREAD

**Discussion by okarlsson** [react-hook-form/react-hook-form#3270] [Q&A]
Hey, really love the lib and it has made it super easy to create complex forms.

I have one thing though that I have gotten stuck on, it comes down to using `watch` together with `useFieldArray`

So this is the scenario. I have a list of skills that i dynamically add or remove from with useFieldArray. But I also for each skill have a circle that fills depending on the value you select, so when you change the value, I want the circle to update.

I seems like i need to `map` on the fieldArray to get everything to work with remove and append, but then it wont re-render to update the indicators.

Anyone who could point me in the correct direction here?

```js
  const { handleSubmit, register, watch, control } = useForm({
    mode: 'onSubmit',
    defaultValues: {  
      skills,
      },
  });

  const { fields: skillsFields, append: appendSkill, remove: removeSkill } = useFieldArray({
    control,
    name: 'skills',
  });

  const watchSkills = watch('skills', skillsFields);

// Does not update when typing in the input
{skillsFields.map((skill, index) => (
   <SkillWrapper key={skill.id}>
     <CircleIndicator value={skill.value} text={skill.name} /> 
     <Input
       name={`skills[${index}].name`}
       ref={register()}
       defaultValue={skill.name} 
      />
      <Select name={`skills[${index}].value`} defaultValue={skill.value} ref={register()}>
        <Option value={1}>Beginner</Option>
        <Option value={2}>Intermediate</Option>
        <Option value={3}>Proficient</Option>
        <Option value={4}>Expert</Option>
       </Select>
    </SkillWrapper>
))}
```

**Comment by bluebill1049** [comment]
can you share a codesandbox?

**Reply by okarlsson** [reply]
Here is a sandbox of the setup I'm looking for: https://codesandbox.io/s/beautiful-agnesi-sd2wp

So what it comes down to is that I want to get the value title to change when I write in the input boxes. I understand that I could use the `watch` function for it, but I can't use it since the title is inside of the map function of the fields

**Comment by mstanielewicz** [comment]
I had the same issue, when I wanted to filter select options, by the values that were already selected.
fields that are returned by useFieldArray are not connected, so it doesn't get updated on select change.

**Reply by bluebill1049** [reply]
can you share a simple codesandbox for this behaviour?

**Reply by mstanielewicz** [reply]
https://codesandbox.io/s/react-hook-form-array-fields-forked-dm7wc?file=/src/ManagersField.js
Here You can see the fields are not updated with values when we select and selectedManagers gets empty when You remove one form the list.

**Reply by bluebill1049** [reply]
thanks, @mstanielewicz what're the steps to reproduce the problem? just select the drop-down and then remove it? what's not expected?

**Reply by mstanielewicz** [reply]
Add for example 3 selects and change value - we get in console:
```
fields: [
  { user: null },
  { user: null },
  { user: null }
]

managers: [
  { user: "1" },
  { user: "2" },
  { user: "3" }
]
```
and when we remove second we end up in console with:
```
fields: [
  { user: "1" },
  { user: "3" }
]

managers: []
```

**Reply by bluebill1049** [reply]
for watch FieldArray: 

`const selectedManagers = form.watch("managers", fields);`

<img width="966" alt="Screen Shot 2020-10-29 at 9 34 43 am" src="https://user-images.githubusercontent.com/10513364/97503882-fbe2d580-19c9-11eb-9983-6663512a0174.png">
